Tests/PrettyPrinterTests/FunctionDeclTests.swift

```diff
@@ -75,8 +79,65 @@ public class FunctionDeclTests: PrettyPrintTestCase {
       print("Hello World")
     }
 
+    func longerNameFun<
+      ReallyLongTypeName: Conform,
+      TypeName
+    >(
+      var1: ReallyLongTypeNAme,
+      var2: TypeName
+    ) {
+      let a = 123
+    }
+
     """
 
     assertPrettyPrintEqual(input: input, expected: expected, linelength: 40)
   }
+
+  public func testFunctionWhereClause() {
+    let input =
+    """
+    public func index<Elements: Collection, Element>(
+      of element: Element,
+      in collection: Elements
+    ) -> Elements.Index? where Elements.Element == Element {
+      let a = 123
+      let b = "abc"
+    }
+
+    public func index<Elements: Collection, Element>(
```

Comment:
It wouldn't hurt to have a separate "end-to-end" test as a separate test method that takes this whole signature and lays it out on one line, just to verify that all the breaks work in unison.